ras_party/controllers/pending_survey_controller.py

```diff
@@ -162,16 +184,46 @@ def accept_share_survey(session, batch_no, new_respondent=None):
             try:
                 with session.begin_nested():
                     enrolment = Enrolment(business_respondent=business_respondent,
-                                          survey_id=pending_share_survey['survey_id'],
+                                          survey_id=pending_survey['survey_id'],
                                           status=EnrolmentStatus.ENABLED)
                     session.add(enrolment)
 
             except SQLAlchemyError as e:
-                logger.exception('Unable to confirm pending share survey', batch_no=batch_no)
+                logger.exception('Unable to confirm pending survey', batch_no=batch_no)
         else:
             logger.info('Ignoring respondent as already enrolled', business_id=business_id, survey_id=survey_id,
-                        email=share_surveys_list[0]['email_address'])
-        delete_share_survey_by_batch_no(batch_no, session)
+                        email=pending_surveys_list[0]['email_address'])
+        delete_pending_survey_by_batch_no(batch_no, session)
+        session.commit()
+        if pending_surveys_is_transfer:
+            try:
+                remove_transfer_originator_business_association(pending_surveys_list[0]['shared_by'])
+            except SQLAlchemyError as e:
```

Comment:
If this fails, how much of the transaction should we be rolling back? At the moment this will leave the previous enrolments loose in the database.

Reply:
I thought about it and wanted to bring everything to the same session
but it wont be possible
as at this point of time the new association has to be present as per the business_respondent constraint, so that I can delete the existing association

Reply:
I have talked to Adam and Gareth and it seems we are ok with this.